scanpopup.cc

```diff
@@ -270,6 +270,9 @@ ScanPopup::ScanPopup( QWidget * parent,
   connect( definition, SIGNAL( titleChanged(  ArticleView *, QString const & ) ),
            this, SLOT( titleChanged(  ArticleView *, QString const & ) ) );
 
+  connect( QApplication::clipboard(), SIGNAL( dataChanged() ),
+           this, SLOT( handleTranslateWordFromClipboard() ) );
```

Comment:
这个方案通用吗？
>void QClipboard::dataChanged()
This signal is emitted when the clipboard data is changed.
>
>On macOS and with Qt version 4.3 or higher, clipboard changes made by other applications will only be detected when the application is activated.

Reply:
This must be breaking the GD popup when the contents of the selection buffer changes (see the connection under `#ifdef HAVE_X11` below).

Reply:
The windows and macOS platforms do not have clipboard issues, so the `translateWordFromClipboardDelayTimer` will do the job. 

The  `QClipboard::dataChanged()` is used to handle the clipboard issue on linux with other Qt programs.

I've considered calling `handleTranslateWordFromClipboard()` directly, but using a timer seems better.

Reply:
I mean the connection you added duplicates and reduces the functionality of the already existing connection just below, because the `dataChanged()` signal is a specialization of the already connected-to `changed( QClipboard::Mode )` signal!

Reply:
I guess `handleTranslateWordFromClipboard()` should have a higher priority, so I just put it into `ScanPopup::clipboardChanged( QClipboard::Mode m )`.

What do you think?